km/km_hcalls.c

```diff
@@ -345,15 +346,16 @@ static int copyin_msghdr(struct msghdr* km_msg, struct msghdr* guest_msg)
    if (km_msg->msg_iovlen > 0) {
       for (int i = 0; i < km_msg->msg_iovlen; i++) {
          // Don't validate addresses if the element has a length of zero.
+         km_msg->msg_iov[i].iov_len = guest_msg->msg_iov[i].iov_len;
```

Comment:
`guest_msg->msg_iov` is a `km_gva_t`. Need to translate to kma before dereferencing. That's why the valgrind test is crashing since is doesn't use GVA == KMA.